### ISSUE TITLE
[BUG] [go] does not compile: missing types: AnyOfintegerstring, AnyOfstringfloat , AnyOfstringinteger, OneOfobjectobject , AnyOfobjectobject ...

### ISSUE THREAD

**Issue by iafilius** [OpenAPITools/openapi-generator#11851]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

While generating library of vendor provider spec file, not all types are generated

AnyOfintegerstring
AnyOfstringfloat
AnyOfstringinteger
AnyOffloat
OneOfobjectobject
AnyOfstringarrayinteger
AnyOfstringboolean
AnyOfobjectobject
AnyOfReplacePatchOperationRemovePatchOperationAddPatchOperationMovePatchOperation
OneOfobjectobjectobject

##### openapi-generator version

Tested openapi generator clie versions 5.0.1, 5.1.1, 5.2.1, (5.[34].x crashes ) 6.0.0-SNAPSHOT (cloned master)

##### OpenAPI declaration file content or url

Full (vendor) version on: https://gist.github.com/iafilius/bbfe04aa5b31d9fd212a7841f71e4447
Essentials version on: https://gist.github.com/iafilius/00c025c50145542684210829bcb9813e

##### Generation Details

$ java -jar ../${CODEGEN} generate \
  -i ${SPECFILE} \
  -g go \
  -o out/ 

##### Steps to reproduce

1. download specfile
2. download recent openapi-generator-cli or build from master
3. generate go libary as described above
4. go to output dir, end try to build: go build in that folder, you may need a few 'go get ..' as indicated go build output
5. watch  errors like:
```
$ go build
# github.com/GIT_USER_ID/GIT_REPO_ID
./model_device_settings_cloud_security.go:20:23: undefined: OneOfobjectobject
./model_device_settings_cloud_security.go:41:65: undefined: OneOfobjectobject
./model_device_settings_cloud_security.go:43:13: undefined: OneOfobjectobject
./model_device_settings_cloud_security.go:51:68: undefined: OneOfobjectobject
./model_device_settings_cloud_security.go:68:65: undefined: OneOfobjectobject
./model_device_settings_ospf_area.go:20:5: undefined: AnyOfintegerstring
./model_device_settings_ospf_area.go:41:42: undefined: AnyOfintegerstring
./model_device_settings_ospf_area.go:43:11: undefined: AnyOfintegerstring
./model_device_settings_ospf_area.go:52:46: undefined: AnyOfintegerstring
./model_device_settings_ospf_area.go:69:42: undefined: AnyOfintegerstring
./model_device_settings_ospf_area.go:43:11: too many errors
```

##### Related issues/PRs

related in go:
https://github.com/OpenAPITools/openapi-generator/issues/11291

related, bug for java:
https://github.com/OpenAPITools/openapi-generator/issues/7246

related for rust:
https://github.com/OpenAPITools/openapi-generator/issues/7975

##### Suggest a fix

Tried to define missing types myself as interface{}, while it compiles, there are issues with nullable/omitempty
It would be nice the missing types would be generated, including the helper functions/methods to marshall etc.

**Comment by blazergame** [comment]
Any movements or workaround for this issue?

**Comment by BKniggeAtomic** [comment]
just noticed this same issue. "undefined: OneOfobjectobject"

**Comment by wing328** [comment]
please try the latest master which has better support for inline schemas.

**Comment by BKniggeAtomic** [comment]
Sorry, I should have mentioned the version number. This was still an issue with 5.4.0 I'm installing and trying the 6.0.0-beta currently.

**Comment by wing328** [comment]
Latest master, not 6.0.0-beta. You can find the snapshot version in the project's readme.

**Comment by BKniggeAtomic** [comment]
Will give that a shot and follow up. 6.0.0-beta had the same issue as well fyi.

**Comment by wing328** [comment]
Right, there are some recent PRs to enhance inline schemas support (after the 6.0.0-beta release).

**Comment by BKniggeAtomic** [comment]
same issue with the latest master.

**Comment by BKniggeAtomic** [comment]
After a bit of googling, what looks like the same bug has been reported and then closed without any resolution multiple times previously.  

This was just the first that I found when initially googling. 

Examples 

https://github.com/OpenAPITools/openapi-generator/issues/1745
https://github.com/OpenAPITools/openapi-generator/issues/2164

**Comment by BKniggeAtomic** [comment]
Seems that this isn't so much a bug as a feature request. Go generators docs state that OneOf is not supported. 

https://openapi-generator.tech/docs/generators/go/#schema-support-feature
